### PR TITLE
Fix debugger initialization

### PR DESCRIPTION
**Pull request by tommcdon**
Fixes debugger initialization regression caused by [#63157](https://github.com/dotnet/runtime/pull/63157)

### CODE REVIEW

File: src/coreclr/vm/domainassembly.h [hunk 1]
```diff
@@ -65,14 +65,16 @@ enum NotificationStatus
 
 class DomainAssembly final
 {
+    VPTR_BASE_CONCRETE_VTABLE_CLASS(DomainAssembly)
```
Comment (jkotas) [inline src/coreclr/vm/domainassembly.h:68]: This is a simple C++ type that nothing inherits from now. It does not need a vtable. This fix seems to be injecting useless vtable into the type.

What is the debugger code that is failing? Should the fix be made in that code instead?

Comment (tommcdon) [reply src/coreclr/vm/domainassembly.h:68]: My goal is to perform a quick fix to unblock the debugger.  The debugger is failing in [DacDbiInterfaceImpl::GetAssemblyFromDomainAssembly](https://github.com/dotnet/runtime/blob/d603356dfbf09616e55bd99ed840aa20468a9e57/src/coreclr/debug/daccess/dacdbiimpl.cpp#L4048) and crashes while handling the assembly load event for System.Private.CoreLib.  I'll create an issue to track removing the v-table.

Comment (jkotas) [reply src/coreclr/vm/domainassembly.h:68]: I think the proper fix is to change `VPTR` to `DPTR` on these lines in vm:

```
typedef VPTR(class DomainAssembly)      PTR_DomainAssembly;
```
```
typedef VPTR(PEAssembly) PTR_PEAssembly;
```

The vtables and DAC are super fragile, and we have multiple workarounds like https://github.com/dotnet/runtime/blob/6dcb03913c7d55dca824abcd21617212bef62553/src/coreclr/vm/frames.h#L354-L360 to make them work on all platforms. If you want to go with this quick fix that adds back the vtables, make sure to test it on all platforms.

Comment (VSadov) [reply src/coreclr/vm/domainassembly.h:68]: I assume there is no way to have a simple test for this?

Comment (tommcdon) [reply src/coreclr/vm/domainassembly.h:68]: > I assume there is no way to have a simple test for this?

CI testing is something we would like to accomplish in the future

**Review by jkotas — APPROVED** [review]

**Comment by tommcdon** [comment]
test failures look unrelated to this PR, merging

**Pull request merged**